registry/debug-adapters.yml

```
@@ -1,101 +1,474 @@
+      port: 3333                                # default value of first gdbserver port
       protocol: swd
       clock: 10000000
+    ui:
+      - section: Debug Interface
+        description: Interface configuration for the debug port
+        options:
+          - name: Clock (kHz)
+            description: Trace configuration
+            yml-node: clock
+            type: number
+            range: [10, 5000]
+            unit: kHz
+            default: 1000kHz
+          - name: Protocol
+            yml-node: protocol
+            type: enum
+            values: [jtag, swd]
+            default: swd
+      - section: Trace
+        description: Trace configuration
+        yml-node: trace
+        select: off
+        options:
+          - name: Clock (kHz)
+            yml-node: trace-clock
+            type: number
+            range: [10, 200000]
+            default: 12000
+            unit: kHz
+          - name: Mode
+            yml-node: trace-port
+            type: enum
+            values: [SWO-UART]
+            default: SWO-UART
+      - section: Telnet
+        description: Telnet server configuration
+        yml-node: telnet
+        select: off
+        options:
+          - name: Port
+            yml-node: port
+            type: number
+            range: [1, 100000]
+            default: 4444
 
   - name: "PICkit@pyOCD"
-    alias-name: ["PICkit Basic"]              # alternative names that map to this debug adapter
-    template: CMSIS-DAP-pyOCD.adapter.json    # template file (initally same as CMSIS-DAP@pyOCD)
-    gdbserver:                                # add gdbserver: node under debugger: in cbuild-run.yml
-    defaults:                                 # default values to use when nowhere specified
-      port: 3333                              # default value of first gdbserver port
+    alias-name: ["PICkit Basic"]                # alternative names that map to this debug adapter
+    template: CMSIS-DAP-pyOCD.adapter.json      # template files  (initally same as CMSIS-DAP@pyOCD)
+    gdbserver:                                  # add gdbserver: node under debugger: in cbuild-run.yml
+    defaults:                                   # default values to use when nowhere specified
+      port: 3333                                # default value of first gdbserver port
       protocol: swd
       clock: 10000000
+    ui:
+      - section: Debug Interface
+        description: Interface configuration for the debug port
+        options:
+          - name: Clock (kHz)
+            description: Trace configuration
+            yml-node: clock
+            type: number
+            range: [10, 5000]
+            unit: kHz
+            default: 1000kHz
+          - name: Protocol
+            yml-node: protocol
+            type: enum
+            values: [jtag, swd]
+            default: swd
+      - section: Trace #not according to ChatGPT
```

Was the comment left on purpose or can be removed?
```suggestion
      - section: Trace
```